OCPVE-230: Improve README content (#333)
openshift/lvm-operator: README.md

```bash
$ export IMG=quay.io/<registry-name>/lvms-operator:latest
$ make docker-build docker-push
```

Ensure that the new repository in your registry is either set to public or that your target OpenShift cluster has read access to that repository. 
Once this is complete, you are ready to proceed with the [deployment steps](#deploying-the-operator).

### Using the pre-built image

If you are comfortable using the prebuilt images, simply set your variable as follows:

```bash
$ export IMG=quay.io/lvms_dev/lvms-operator:latest
```

### Deploying the operator

You can begin the deployment by running the following command:

```bash
$ make deploy
```

After the previous command has completed successfully, switch over to the `openshift-storage` namespace:

```bash
$ oc project openshift-storage
```

Wait until all Pods have started running:

```bash
$ oc get pods -w
```

Once all Pods are running, create a sample LVMCluster CR:

```bash
$ oc create -n openshift-storage -f https://github.com/openshift/lvm-operator/raw/main/config/samples/lvm_v1alpha1_lvmcluster.yaml
```

Upon deployment of the Custom Resource (CR), the following actions will be executed:

> - A Logical Volume Manager (LVM) volume group named `vg1` will be created, utilizing all available disks on the cluster.

Reviewer: ```suggestion
- A Logical Volume Manager (LVM) volume group named `vg1` is created, utilizing all available disks on the cluster.
```

> - A thin pool named `thin-pool-1` will be created within `vg1`, with a size equivalent to 90% of `vg1`.

Reviewer: ```suggestion
- A thin pool named `thin-pool-1` is created within `vg1`, with a size equivalent to 90% of `vg1`.
```

> - The TopoLVM Container Storage Interface (CSI) plugin will be deployed, resulting in the launch of the `topolvm-controller` and `topolvm-node` Pods.

Reviewer: ```suggestion
- The TopoLVM Container Storage Interface (CSI) plugin is deployed, resulting in the launch of the `topolvm-controller` and `topolvm-node` Pods.
```

> - A Storage Class and a Volume Snapshot Class will be created, both named `lvms-vg1`, to facilitate storage provisioning for OpenShift workloads. 

Reviewer: ```suggestion
- A Storage Class and a Volume Snapshot Class are created, both named `lvms-vg1`. This facilitates storage provisioning for OpenShift workloads. 
```

> The Storage Class will have the `WaitForFirstConsumer` volume binding mode that is utilized in a multi-node configuration to optimize the scheduling of pod placement.

Reviewer: ```suggestion
The Storage Class is configured with the `WaitForFirstConsumer` volume binding mode that is utilized in a multi-node configuration to optimize the scheduling of pod placement.
```

> This strategy prioritizes the allocation of Pods to Nodes with the greatest amount of available storage capacity.

Reviewer: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-references

```suggestion
This strategy prioritizes the allocation of pods to nodes with the greatest amount of available storage capacity.
```

> - The LVMS system also creates two additional internal Custom Resources to support its functionality. A Custom Resource called LVMVolumeGroup that is generated and managed by LVMS to monitor the individual Volume Groups across multiple Nodes in the cluster.

Reviewer: ```suggestion
- The LVMS system also creates two additional internal CRs to support its functionality:
* `LVMVolumeGroup` is generated and managed by LVMS to monitor the individual Volume Groups across multiple nodes in the cluster.
* `LVMVolumeGroupNodeStatus` is created by the VG Manager. This CR is used to monitor the status of volume groups on individual nodes in the cluster.
```

>   The second internal Custom Resource created by the VG Manager is known as LVMVolumeGroupNodeStatus. This Custom Resource is responsible for monitoring the status of volume groups on individual nodes within the cluster.

Reviewer: remove this line

Wait until the LVMCluster reaches the `Ready` status:

```bash
$ oc get lvmclusters.lvm.topolvm.io my-lvmcluster
NAME            STATUS
my-lvmcluster   Ready
```

> Check all Pods again and wait until they are all running:

Reviewer: ```suggestion
Wait until all pods are active:
```

```bash
$ oc get pods -w
```

> The `topolvm-node` Pod will remain in the initialization phase until the `vg-manager` completes all the necessary preparations, which may take some time.

Reviewer: ```suggestion
The `topolvm-node` pod remains in the initialization phase until the `vg-manager` completes all the necessary preparations.
```

Once all the Pods have been launched, the LVMS is ready to manage your Logical Volumes and make them available for use in your applications.

### Inspecting the storage objects on the Node

Prior to the deployment of the Logical Volume Manager Storage (LVMS), there are no pre-existing LVM Physical Volumes (PVs), Volume Groups (VGs), or Logical Volumes (LVs) associated with the disks.

```bash
sh-4.4# lsblk
NAME    MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sdb       8:16   0 893.8G  0 disk
|-sdb1    8:17   0     1M  0 part
|-sdb2    8:18   0   127M  0 part
|-sdb3    8:19   0   384M  0 part /boot
`-sdb4    8:20   0 893.3G  0 part /sysroot
sr0      11:0    1   987M  0 rom
nvme0n1 259:0    0   1.5T  0 disk
nvme1n1 259:1    0   1.5T  0 disk
nvme2n1 259:2    0   1.5T  0 disk
sh-4.4# pvs
sh-4.4# vgs
sh-4.4# lvs
```

> Upon successful completion of the deployment process, the necessary LVM Physical Volumes (PVs), Volume Groups (VGs), and Thin Pools will be created on the host.

Reviewer: ```suggestion
After successful deployment, the necessary LVM physical volumes, volume groups, and thin pools are created on the host.
```

```bash
sh-4.4# pvs
  PV           VG  Fmt  Attr PSize  PFree 
  /dev/nvme0n1 vg1 lvm2 a--  <1.46t <1.46t
  /dev/nvme1n1 vg1 lvm2 a--  <1.46t <1.46t
  /dev/nvme2n1 vg1 lvm2 a--  <1.46t <1.46t
sh-4.4# vgs
  VG  #PV #LV #SN Attr   VSize  VFree 
  vg1   3   0   0 wz--n- <4.37t <4.37t
sh-4.4# lvs
  LV          VG  Attr       LSize  Pool Origin Data%  Meta%  Move Log Cpy%Sync Convert
  thin-pool-1 vg1 twi-a-tz-- <3.93t             0.00   1.19      
```

### Testing the operator

> Once you completed [the deployment steps](#deploying-the-operator), you can proceed to create a basic test application that will consume storage.

Reviewer: ```suggestion
Once you have completed [the deployment steps](#deploying-the-operator), you can proceed to create a basic test application that will consume storage.
```

    ```bash
    $ export IMG=quay.io/<registry-name>/lvms-operator:latest
    ```

2. Build and push the container image:

    ```bash
    $ make docker-build docker-push
    ```

Ensure that the OpenShift cluster has read access to that repository. Once this is complete, you are ready to proceed with the [deployment steps](#deploying-the-operator).

### Using the pre-built image

If you are comfortable using the pre-built images, simply set your variable as follows:

```bash
$ export IMG=quay.io/lvms_dev/lvms-operator:latest
```

### Deploying the operator

You can begin the deployment by running the following command:

```bash
$ make deploy
```

After the previous command has completed successfully, switch over to the `openshift-storage` namespace:

```bash
$ oc project openshift-storage
```

Wait until all pods have started running:

```bash
$ oc get pods -w
```

Once all pods are running, create a sample `LVMCluster` custom resource (CR):

```bash
$ oc create -n openshift-storage -f https://github.com/openshift/lvm-operator/raw/main/config/samples/lvm_v1alpha1_lvmcluster.yaml
```

> Upon deployment of the CR, the following actions are executed:

Reviewer: ```suggestion
After the CR is deployed, the following actions are executed:
```
